File: 2016/nazionali/aeroporto/att/grader.pas
uses aeroporto;

var	
	fr, fw : text;

{ Declaring variables }
	R : Longint;
	A : array of Longint;
	B : array of Longint;
	T : array of Longint;

{ iterators used in for loops }
	i0: LongInt;

begin
{$ifdef EVAL}
    assign(fr, 'input.txt');
    assign(fw, 'output.txt');
{$else}
    fr := input;
    fw := output;
{$endif}
    reset(fr);
    rewrite(fw);

	{ Reading input }
	read(fr, R);
	Setlength(A, R);
	Setlength(B, R);
	for i0 := 0 to R-1 do
	begin
		read(fr, A[i0]);
		read(fr, B[i0]);
	end;

	{ Calling functions }
	Setlength(T, R);
	pianifica(R, A, B, T);

	{ Writing output }
	for i0 := 0 to R-1 do
	begin
		write(fw, T[i0], ' ');
	end;
	writeln(fw);
	
	close(fr);
    close(fw);
end.

File: 2016/nazionali/aeroporto/att/grader.c
#include <stdio.h>
#include <assert.h>
#include <stdlib.h>

static FILE *fr, *fw;

// Declaring variables
static int R;
static int* A;
static int* B;
static int* T;

// Declaring functions
void pianifica(int R, int* A, int* B, int* T);

int main() {
	#ifdef EVAL
		fr = fopen("input.txt", "r");
		fw = fopen("output.txt", "w");
	#else
		fr = stdin;
		fw = stdout;
	#endif

	// Iterators used in for loops
	int i0;

	// Reading input
	fscanf(fr, "%d ", &R);
	A = (int*)malloc(R * sizeof(int));
	B = (int*)malloc(R * sizeof(int));
	for (i0 = 0; i0 < R; i0++) {
		fscanf(fr, "%d %d ", &A[i0], &B[i0]);
	}

	// Calling functions
	T = (int*)malloc(R * sizeof(int));
	pianifica(R, A, B, T);

	// Writing output
	for (i0 = 0; i0 < R; i0++) {
		fprintf(fw, "%d ", T[i0]);
	}
	fprintf(fw, "\n");
	
	fclose(fr);
	fclose(fw);
	return 0;
}
